modules/dnn/test/test_layers_1d.cpp

```diff
@@ -245,4 +245,37 @@ INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Elemwise_1d_Test, Combine(
 /*operation*/           Values("div", "prod", "max", "min", "sum")
 ));
 
+typedef testing::TestWithParam<tuple<std::vector<int>>> Layer_Const_Test;
+TEST_P(Layer_Const_Test, Accuracy)
+{
+    std::vector<int> input_shape = get<0>(GetParam());
+
+    LayerParams lp;
+    lp.type = "Const";
+    lp.name = "ConstLayer";
+
+    Mat constBlob, output_ref;
+    if (input_shape.size() > 1)
+        constBlob = Mat(input_shape, CV_32F);
+    else
+        constBlob = Mat(input_shape[0], input_shape.data(), CV_32F);
+    cv::randn(constBlob, 0.0, 1.0);
+    output_ref = constBlob.clone();
+
+    lp.blobs.push_back(constBlob);
+    Ptr<Layer> layer = ConstLayer::create(lp);
+
+    std::vector<Mat> inputs; // No inputs are needed for a ConstLayer
+    std::vector<Mat> outputs;
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
```

Comment:
It is redundant. Shapes are already checked in `normAssert`.

Reply:
Unresolved comment @Abdurrahheem

Reply:
This is not redundant. normAssert does not check for shapes, and for that reason was an explicit check introduced. Otherwise layer might outputs incorrect shape